scripts/build-platypus-csd3-ampere.sh

```diff
@@ -0,0 +1,274 @@
+    # This is just until MFEM merges Edward's changes. Without this, GPU build crashes!
+    git checkout EdwardPalmer99/add-missing-header-to-exodus-writer-fix
+    mkdir build
+    cd build || exit 1
+    echo "Building MFEM"
+    cmake .. \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_SHARED_LIBS=YES \
+        -DMFEM_USE_OPENMP=NO \
+        -DMFEM_THREAD_SAFE=YES \
+        -DMFEM_ENABLE_EXAMPLES=YES \
+        -DMFEM_ENABLE_MINIAPPS=YES \
+        -DMFEM_USE_MPI=YES \
+        -DMFEM_USE_CUDA=YES \
+        -DCUDA_ARCH=sm_80 \
+        -DMFEM_USE_METIS_5=YES \
+        -DMFEM_USE_SUPERLU=YES \
+        -DMFEM_USE_NETCDF=YES \
+        -DMFEM_USE_GSLIB=YES \
+        -DGSLIB_DIR="${BUILD_PATH}/gslib/build" \
+        -DSuperLUDist_DIR="${SLU_DIR}" \
+        -DSuperLUDist_VERSION_OK=YES
+
+    if [ $? -eq 2 ]; then
+        echo "MFEM config failed"
+        exit 1
+    fi
+
+    make -j"$compile_cores"
+
+    if [ $? -eq 2 ]; then
+        echo "MFEM build failed"
+        exit 1
+    fi
+
+    LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${BUILD_PATH}/mfem/build:${BUILD_PATH}/mfem/build/miniapps/common
+}
+
+install_moose() {
+
+    # Some of the variables needed
+    export MOOSE_JOBS=$compile_cores
+    export LIBMESH_JOBS=$compile_cores
+    export METHOD="opt"
+    SLEPC_DIR=$(spack find --format "{prefix}" slepc arch=${ARCH})
+    export SLEPC_DIR
+
+    cd "${BUILD_PATH}" || exit 1
+    git clone https://github.com/idaholab/moose
+    cd moose || exit 1
+
+    echo "Building libmesh..."
+    ./scripts/update_and_rebuild_libmesh.sh --with-mpi
+    if [ $? -eq 2 ]; then
+        echo "libmesh build failed"
+        exit 1
+    fi
+
+    echo "Building WASP..."
+    ./scripts/update_and_rebuild_wasp.sh
+    if [ $? -eq 2 ]; then
+        echo "WASP build failed"
+        exit 1
+    fi
+
+    ./configure --with-derivative-size=200
+    if [ $? -eq 2 ]; then
+        echo "MOOSE configure failed"
+        exit 1
+    fi
+
+    cd framework || exit 1
+    make -j"$compile_cores"
+    if [ $? -eq 2 ]; then
+        echo "MOOSE framework build failed"
+        exit 1
+    fi
+
+    cd ../modules || exit 1
+    make -j"$compile_cores"
+    if [ $? -eq 2 ]; then
+        echo "MOOSE modules build failed"
+        exit 1
+    fi
+
+    # This takes very long! Only run the tests if you really need to!
+    #cd ../test || exit 1
+    #make -j"$compile_cores"
+    #if [ $? -eq 2 ]; then
+    #    echo "MOOSE test build failed"
+    #    exit 1
+    #fi
+
+    #./run_tests -j"$compile_cores"
+}
+
+install_platypus() {
+
+    cd "${BUILD_PATH}" || exit 1
+
+    echo "Building platypus..."
+    git clone https://github.com/aurora-multiphysics/platypus.git
+    cd platypus || exit 1
+    git submodule update --init --recursive
+    cd contrib/hephaestus/ || exit 1
+    mkdir build
+    cd build || exit 1
+    cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DMFEM_DIR="${BUILD_PATH}/mfem/build" ..
+    ninja
+    cd "${BUILD_PATH}"/platypus || exit 1
```

Comment:
The `hephaestus` submodule no longer exists in `platypus` - this section causes an error, as it attempts to `cd` into a non-existent directory. Should just build platypus here.